src/Assets.php

```diff
@@ -60,37 +60,43 @@ public static function register_assets() {
 			$asset_api->register_script( 'wc-blocks-checkout', 'build/blocks-checkout.js', [] );
 		}
 
-		// Pass in settings specific to blocks.
-		$product_counts = wp_count_posts( 'product' );
-
-		$asset_data_registry->add( 'wcBlocksPluginUrl', plugins_url( '/', __DIR__ ) );
-		$asset_data_registry->add( 'wcBlocksPhase', Package::feature()->get_flag() );
-
-		/*
-		 * translators: If your word count is based on single characters (e.g. East Asian characters),
-		 * enter 'characters_excluding_spaces' or 'characters_including_spaces'. Otherwise, enter 'words'.
-		 * Do not translate into your own language.
-		 */
-		$asset_data_registry->add( 'wordCountType', _x( 'words', 'Word count type. Do not translate!', 'woo-gutenberg-products-block' ) );
-		$asset_data_registry->add( 'placeholderImgSrc', wc_placeholder_img_src() );
-		$asset_data_registry->add( 'isLargeCatalog', $product_counts->publish > 100 );
-		$asset_data_registry->add( 'productCount', array_sum( (array) $product_counts ) );
-		$asset_data_registry->add(
-			'restApiRoutes',
-			[
-				'/wc/store' => array_keys( Package::container()->get( RestApi::class )->get_routes_from_namespace( 'wc/store' ) ),
-			]
+		// Pass extra data to the client from the server, based on script handle.
+		wp_add_inline_script(
+			'wc-blocks-config',
+			"
+			var wcBlocksConfig = {
+				pluginUrl: '" . esc_js( plugins_url( '/', __DIR__ ) ) . "',
+				buildPhase: '" . esc_js( Package::feature()->get_flag() ) . "'
+			};
+			",
+			'before'
+		);
```

Comment:
It's still not clear to me in this case why this data needs to be passed through to the client this way. You could accomplish the same here by just doing the following in `AbstractBlock::enqueue_data`:

```suggestion
		$aset_data_registry->add(
			'wcBlocksConfig',
			[
			    'pluginUrl' => plugins_url( '/', __DIR__ ),
			    'buildPhase' =>  Package::feature()->get_flag()
			]
		);
```

That lets the asset data registry take care of escaping and everything for you? Then because it's in `AbstractBlock::enqueue_data` the resulting json object will only be available on a route that has a block in it?

Reply:
I'm concerned that these build variables are used everywhere - are we certain block types cover all possible routes?

Reply:
Well, we currently have all our blocks registered server side right and as far as I can tell every block at a minimum extends `AbstractBlock` so that means `enqueue_data` is going to be fired server side for every registered block?

Your concern is definitely noted and important to consider though if we have any of our JavaScript running on routes that might not have a block (or block components)?